spring-test/src/main/java/org/springframework/test/context/support/AbstractContextLoader.java

```diff
@@ -132,6 +135,9 @@ protected void prepareContext(ConfigurableApplicationContext context, MergedCont
 		TestPropertySourceUtils.addPropertiesFilesToEnvironment(context, mergedConfig.getPropertySourceLocations());
 		TestPropertySourceUtils.addInlinedPropertiesToEnvironment(context, mergedConfig.getPropertySourceProperties());
 		invokeApplicationContextInitializers(context, mergedConfig);
+		for (ContextCustomizer contextCustomizer : mergedConfig.getContextCustomizers()) {
+			contextCustomizer.prepareContext(context, mergedConfig);
+		}
```

Comment:
I'm not very comfortable having a `ContextCustomizer` being called from a `prepareContext()` method when a `customizeContext()` method exists as well.